scripts/jenkins/Jenkinsfile

```diff
@@ -1,5 +1,33 @@
 platforms = ["centos7", "debian9", "ubuntu16", "ubuntu18"]
 
+/*  The detectChanges function looks at the last git commit and searches for changes to
+*   "redhat/freeradius.spec" or anything in the "debian" directory.  If it 
+*   finds any changes to those files it returns the string " --no-cache " 
+*   which can be fed to a docker build command to tell docker to rebuild 
+*   the image from scratch so that it can pick up any new dependencies 
+*   specified in those package manifest files
+*/
+
+def detectChanges() {
+    def string = ""
+    def changeLogSets = currentBuild.changeSets
+    for (int i = 0; i < changeLogSets.size(); i++) {
+        def entries = changeLogSets[i].items
+        for (int j = 0; j < entries.length; j++) {
+            def entry = entries[j]
+            def files = new ArrayList(entry.affectedFiles)
+            for (int k = 0; k < files.size(); k++) {
+                def file = files[k]
+                if (file.path =~ /(^debian\/.*)|(redhat\/freeradius.spec)/) {
```

Comment:
@mattrose Probably you should do the same test as is doing for Debian due to existing others files into Redhat/ directory instead of only the `*.spec`

Reply:
The freeradius.spec file is the only file that depends on the environment it's being built in.  All other files in that directory are patches or ancillary files that get added to the package as it's being built, and we don't really want to rebuild the whole image if there's only a change in them.  I would have made a similar change and only rebuilt on modifications to `debian/control.in` but I'm not as sure of the other files in that directory as I am of the redhat directory